### ISSUE TITLE
Can't use keyword as item-fn

### ISSUE THREAD

**Issue by jcf** [bluemont/con-hash#1]
There's a pre clause that requires `item-fn` to be a function, but in the case of maps a keyword may often be appropriate. As a workaround, I can create an anonymous function but it'd be great if a keyword could be passed.

``` clj
(consistent-hash-fn :db/id identity) ; throws an exception
(consistent-hash-fn #(get % :db/id) identity) ; is okay
```

I can probably put together a PR this evening if you're open to accepting the change.

**Comment by jcf** [comment]
Eek. Distribution isn't great with a range from 0 to 100! :smile:

**Comment by xpe** [comment]
@jcf Good idea. Changing the precondition to allow either a function or keyword would work.
